### ISSUE TITLE
Alt-clicking a blood decal tile may display 150+ different blood icons at once

### ISSUE THREAD

**Issue by Angustmeta** [tgstation/tgstation#53824]
## Round ID:
[146560](https://scrubby.melonmesa.com/round/145348) Terry. 

## Testmerges:
#53596
#53329

## Reproduction:
Not sure how to reproduce, but this is what I did:

1. Surrounded a tile with ATMOS holofan projector fans. 
2. Used an RCD to remove the plating of the middle tile, making a hole to space. 
3. Constructed half tile rwindows to surround the hole. 
4. Alt-clicked one of the holofan tiles to be able to click the rwindow with screwdriver. It had a blood decal on it. 

Not sure if any of the above matters at all. 

**Result:** 
![200918 1940 15](https://user-images.githubusercontent.com/46400996/93631748-6b899c00-f9ec-11ea-9919-7c2cb2ca554f.png)

After that, alt-clicking any bloody tile on station (but not in the CTF arena) would make it look the same. 

![200918 1944 44](https://user-images.githubusercontent.com/46400996/93631849-95db5980-f9ec-11ea-81a2-bd54f13234b8.png)

**Comment by Couls** [comment]
this might be an asset cache related thing, i'll look into it

**Comment by Couls** [comment]
hopped on bagil and tried to reproduce, no luck 
![image](https://user-images.githubusercontent.com/30060146/93671413-d4b2f300-fa70-11ea-9a2e-726720212ec4.png)
I'll try those exact steps on local next time I boot up my test server

**Comment by Angustmeta** [comment]
Yeah I'll post again if I manage to reproduce somehow. Maybe it has to do with alt-clicking only a few pixels away from the space tile or something.

**Comment by timothymtorres** [comment]
It seems that whatever was causing this had several issues reported around the same time.  I cannot find any recent issues involving this bug... I suspect it might have been a BYOND issue that got fixed somewhere by Lummox.

@Mothblocks what do you reckon?

**Comment by MrStonedOne** [comment]
This happens with the cdn asset system (ever since the verb panel became a browser window, it uses this).

The bug still exists and in fact mothblocks made it worse with a recent (ish) pr to icon2html

It happens any time an object has an icon state not defined by the dmi. such cases result in the dmi getting sent instead of just one icon state, where as normal byond behavior is to use a blank iconstate or the first icon state, depending

**Comment by Mothblocks** [comment]
what was my recent (ish) pr to icon2html i dont think ive touched that in a year

**Comment by Mothblocks** [comment]
are you talking about #58607

**Comment by MrStonedOne** [comment]
oh, it was the fruit!

@LemonInTheDark

**Comment by LemonInTheDark** [comment]
All my accomplishments turned to mothfluff. 😢 

I've not checked directly, but I have a feeling most of the cost there was icon_states() doing a file read or something? cause some of the access times were wayyy out of wack. May just be people being incapable of keeping dmi sizes low though.

**Comment by LemonInTheDark** [comment]
today is a good day

**Comment by LemonInTheDark** [comment]
tho hm, if icon2html still exists the core problem here is still one. @MrStonedOne @jlsnow301 thoughts?
